baseapp/baseapp.go

```diff
@@ -942,7 +939,7 @@ func (app *BaseApp) runTx(ctx sdk.Context, mode runTxMode, txBytes []byte) (gInf
 		// NOTE: Alternatively, we could require that AnteHandler ensures that
 		// writes do not happen if aborted/failed.  This may have some
 		// performance benefits, but it'll be more difficult to get right.
-		anteCtx, msCache = app.cacheTxContext(ctx, txBytes)
+		anteCtx, msCache = app.cacheTxContext(ctx, checksum)
```

Comment:
Can you explain why we use the SHA256 over the txBytes?

Reply:
@jewei1997 in `cacheTxContext`, `txBytes` is only used to calculate its sha256, so we just pass in the calculated sha256 directly